src/FieldsBuilder.php

```diff
@@ -824,4 +853,220 @@ private function getPrefetchMethodInfo(ReflectionClass $refClass, $reflector, ob
+                    } catch (CannotMapTypeExceptionInterface $e) {
+                        $e->addReturnInfo($refMethod);
+                        throw $e;
+                    }
+                } else {
+                    $type = $args[$name]->getType();
+                    if ($isUpdate && $type instanceof NonNull) {
+                        $type = $type->getWrappedType();
+                    }
+                }
+
+                $inputFieldDescriptor->setHasDefaultValue($isUpdate);
+                $inputFieldDescriptor->setDefaultValue($args[$name]->getDefaultValue());
+                $constructerParameters = $this->getClassConstructParameterNames($refClass);
+                if(!in_array($name, $constructerParameters)) {
+                    $inputFieldDescriptor->setTargetMethodOnSource($methodName);
+                }
+
+                $inputFieldDescriptor->setType($type);
+                $inputFieldDescriptor->setInjectSource($injectSource);
+
+                $inputFieldDescriptor->setMiddlewareAnnotations($this->annotationReader->getMiddlewareAnnotations($refMethod));
+
+                $field = $this->inputFieldMiddleware->process($inputFieldDescriptor, new class implements InputFieldHandlerInterface {
+                    public function handle(InputFieldDescriptor $inputFieldDescriptor): ?InputField
+                    {
+                        return InputField::fromFieldDescriptor($inputFieldDescriptor);
+                    }
+                });
+
+
+
+
+                if ($field === null) {
+                    continue;
+                }
+
+                $fields[$name] = $field;
+            }
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Gets input fields by class property annotations.
+     *
+     * @param string|object $controller
+     * @param class-string<AbstractRequest> $annotationName
+     * @param array<mixed> $defaultProperties
+     *
+     * @return array<string, InputField>
+     *
+     * @throws AnnotationException
+     */
+    private function getInputFieldsByPropertyAnnotations($controller, ReflectionClass $refClass, ReflectionProperty $refProperty, string $annotationName, array $defaultProperties, ?string $typeName = null, bool $isUpdate = false): array
+    {
+        $fields = [];
+
+        $annotations = $this->annotationReader->getPropertyAnnotations($refProperty, $annotationName);
+        $docBlock = $this->cachedDocBlockFactory->getDocBlock($refProperty);
+        foreach ($annotations as $annotation) {
+            $description = null;
+
+            if ($annotation instanceof Field) {
+                $for = $annotation->getFor();
+                if ($typeName && $for && !in_array($typeName, $for)) {
+                    continue;
+                }
+
+                $description = $annotation->getDescription();
+
+
+                $name = $annotation->getName() ?: $refProperty->getName();
```

Comment:
Wouldn't the null coalescing operator be more appropriate here?  `??`

Reply:
its also just copied from the equivalent code for output types, might have been a backwards compatible reason for it.

i think it should probably be updated in the whole library at some point, maybe using rector to automate it.